File: Contest 44 Quarter #2/Q1 1467. 询问冷却时间/li.py
class Solution:
    """
    @param arr: The release order
    @param n: The cooldown
    @return: Return the time
    """
    def askForCoolingTime(self, arr, n):
        time = {i: -n for i in set(arr)}
        cur = 0
        for i in arr:
            cur = time[i] = max(time[i] + n, cur) + 1
        return cur

############ test cases ###########
s = Solution()
print s.askForCoolingTime(arr=[1, 1, 2, 2], n=2)
print s.askForCoolingTime(arr=[1, 2, 1, 2], n=2)

File: Contest 44 Quarter #2/Q1 1467. 询问冷却时间/li.java
import java.util.*;

class Solution {
    /**
     * @param arr: The release order
     * @param n: The cooldown
     * @return: Return the time
     */
    public int askForCoolingTime(int[] arr, int n) {
        // Write your code here
        HashMap<Integer, Integer> m = new HashMap<>();
        int cur = 0;
        for (int i : arr) {
            cur  = Math.max(m.getOrDefault(i, -n) + n, cur) + 1;
            m.put(i, cur);
        }
        return cur;
    }
}

/********************* test case *********************/

class Main {
    public static void main(String[] args) {
        Solution s = new Solution();
    }
}